### PR TITLE
Add lowering for the `lapack_dsyevd` custom call

### PR DESCRIPTION
**Pull request by maliasadi**
This lowering will enable the QJIT compilation of `qml.BlockEncode` and `qml.qsvt` with variable matrices.

For example, the compilation of the following circuits are now possible

``` python
@qml.qjit
@qml.qnode(dev)
def circuit1(A):
    qml.BlockEncode(A, wires=range(2))
    return qml.state()


>>> circuit1(np.array([[0.1, 0.2], [0.3, 0.4]]))
``` 

``` python
A = np.array([[0.1, 0.2], [0.3, 0.4]])
angles = np.array([0.1, 0.2, 0.3])

@qml.qjit
@qml.qnode(dev)
def circuit2(A):
    qml.qsvt(A, angles, wires=[0, 1])
    return qml.expval(qml.PauliZ(wires=0))

circuit2(A)
```

[sc-54971]
[sc-55370]

### CODE REVIEW

File: mlir/lib/Catalyst/Transforms/catalyst_to_llvm.cpp [hunk 1]
```diff
@@ -319,12 +319,13 @@ struct CustomCallOpPattern : public OpConversionPattern<CustomCallOp> {
...
 
         Type voidType = LLVM::LLVMVoidType::get(ctx);
-        auto type = LLVM::LLVMFunctionType::get(voidType, {/*args=*/ptr, /*rets=*/ptr});
         auto point = rewriter.saveInsertionPoint();
         ModuleOp mod = op->getParentOfType<ModuleOp>();
         rewriter.setInsertionPointToStart(mod.getBody());
-        LLVM::LLVMFuncOp customCallFnOp =
-            rewriter.create<LLVM::LLVMFuncOp>(loc, op.getCallTargetName(), type);
+
+        LLVM::LLVMFuncOp customCallFnOp = mlir::LLVM::lookupOrCreateFn(
```
Comment (rmoyard) [inline mlir/lib/Catalyst/Transforms/catalyst_to_llvm.cpp:326]: Good catch :+1:

**Review by rmoyard — COMMENTED** [review]
Thanks for adding this function, it looks good :+1: Could add an integration test that would call this function and check for correctness?

File: frontend/catalyst/utils/libcustom_calls.cpp [hunk 2]
```diff
@@ -50,7 +51,10 @@ void dgesdd_(char *jobz, lapack_int *m, lapack_int *n, double *a, lapack_int *ld
              double *u, lapack_int *ldu, double *vt, lapack_int *ldvt, double *work,
              lapack_int *lwork, lapack_int *iwork, lapack_int *info);
 
-// Wrapper to call the SVD solver dgesdd_ from Lapack:
+void dsyevd_(char *jobz, char *uplo, lapack_int *n, double *a, int *lda, double *w, double *work,
+             lapack_int *lwork, lapack_int *iwork, lapack_int *liwork, lapack_int *info);
+
+// Wrapper to call the SVD solver dgesdd_ and dsyevd_ from Lapack:
```
[the remark this answers is not in the document]
Comment (maliasadi) [reply frontend/catalyst/utils/libcustom_calls.cpp]: Added [7aef6d3](https://github.com/PennyLaneAI/catalyst/pull/488/commits/7aef6d3720092b8fa99f8509234fe11089ce2f97) with fixing the format.

**Review by rmoyard — APPROVED** [review]
Nice :100:

File: doc/changelog.md [hunk 3]
```diff
@@ -170,6 +170,23 @@ would no longer decompose the `PSWAP` and `ISWAP` gates to `SWAP`s, `CNOT`s and
...
+  ```python
+  @qml.qjit
+  def workflow(A):
+      B = qml.math.sqrt_matrix(A)
+      return B @ A
+  ```
+
+* Fix the issue with multiple lapack symbol definitions in the compiled program by updating
+  the `stablehlo.custom_call` conversion pass.
+  [(#488)](https://github.com/PennyLaneAI/catalyst/pull/488)
```
Comment (dime10) [inline doc/changelog.md:189]: I don't see a test case for this bug fix, can we add one? :)

Comment (maliasadi) [reply doc/changelog.md:189]: Ah! sure, I have it in my other branch w/ qml.BlockEncode tests. let me add a simple test case in this PR too. :+1:

**Pull request merged**